### PR TITLE
Add support for keychain access flags

### PR DESCRIPTION
**Pull request by xieweizhi**

### CODE REVIEW

File: Sources/KeychainSwift.swift [hunk 1]
```diff
@@ -95,13 +97,28 @@ open class KeychainSwift {
     let accessible = access?.value ?? KeychainSwiftAccessOptions.defaultOption.value
       
     let prefixedKey = keyWithPrefix(key)
+    var accessWithFlag:Any = accessible as Any
```
Comment (ArtemGoryunov) [inline Sources/KeychainSwift.swift:100]:
```suggestion
    let accessWithFlag: Any = accessible as Any
```

I think you can do 
```
let accessWithFlag: Any
```
and then
```
if (..) {
accessWithFlag = SecAccessControlCreateWithFlags(nil, accessible as CFString, control.value, nil) as Any
} else {
accessWithFlag = accessible as Any
}
```

File: Sources/KeychainSwift.swift [hunk 2]
```diff
@@ -95,13 +97,28 @@ open class KeychainSwift {
...
+    
+    var query: [String : Any]
```
Comment (ArtemGoryunov) [inline Sources/KeychainSwift.swift:102]:
```suggestion
    let query: [String : Any]
```

File: Sources/KeychainSwift.swift [hunk 3]
```diff
@@ -95,13 +97,28 @@ open class KeychainSwift {
...
+      
+    if let control = controlFlag{
```
Comment (ArtemGoryunov) [inline Sources/KeychainSwift.swift:104]:
```suggestion
    if let control = controlFlag {
```

File: Sources/KeychainSwift.swift [hunk 4]
```diff
@@ -95,13 +97,28 @@ open class KeychainSwift {
...
+        accessWithFlag = SecAccessControlCreateWithFlags(nil, accessible as CFString, control.value, nil) as Any
+        query = [
+        KeychainSwiftConstants.klass       : kSecClassGenericPassword,
+        KeychainSwiftConstants.attrAccount : prefixedKey,
+        KeychainSwiftConstants.valueData   : value,
+        KeychainSwiftConstants.accessControl  : accessWithFlag
+        ]
+    }else{
```
Comment (ArtemGoryunov) [inline Sources/KeychainSwift.swift:112]:
```suggestion
    } else {
```

File: Sources/KeychainSwift.swift [hunk 5]
```diff
@@ -125,12 +142,12 @@ open class KeychainSwift {
...
   @discardableResult
   open func set(_ value: Bool, forKey key: String,
-    withAccess access: KeychainSwiftAccessOptions? = nil) -> Bool {
+    withAccess access: KeychainSwiftAccessOptions? = nil, withControlFlag controlFlag: KeychainSwiftAccessControlFlag? = nil) -> Bool {
   
     let bytes: [UInt8] = value ? [1] : [0]
     let data = Data(bytes)
 
-    return set(data, forKey: key, withAccess: access)
+    return set(data, forKey: key, withAccess: access,withControlFlag: controlFlag)
```
Comment (ArtemGoryunov) [inline Sources/KeychainSwift.swift:150]:
```suggestion
    return set(data, forKey: key, withAccess: access, withControlFlag: controlFlag)
```

**Review by ArtemGoryunov — APPROVED** [review]

File: Sources/KeychainSwift.swift (hunk 4 above)
Comment (xieweizhi) [reply Sources/KeychainSwift.swift]: Fixed in: 067bb33c34c39595b6b117e014665a5c661bfca4

File: Sources/KeychainSwift.swift (hunk 5 above)
Comment (xieweizhi) [reply Sources/KeychainSwift.swift]: Fixed in: 067bb33c34c39595b6b117e014665a5c661bfca4

File: Sources/KeychainSwift.swift (hunk 3 above)
Comment (xieweizhi) [reply Sources/KeychainSwift.swift]: Fixed in: 067bb33c34c39595b6b117e014665a5c661bfca4

File: Sources/KeychainSwift.swift (hunk 2 above)
Comment (xieweizhi) [reply Sources/KeychainSwift.swift:101]: Query should be `var`, before the function return, query will be updated again.

File: Sources/KeychainSwift.swift (hunk 1 above)
Comment (xieweizhi) [reply Sources/KeychainSwift.swift]: Fixed in: 067bb33c34c39595b6b117e014665a5c661bfca4

**Pull request merged**